src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double AMMOUT_OF_INCREASE = 1.5;
+    private Object[] array;
+    private int currentSize;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (currentSize == array.length) {
+            increaseSize();
+        }
```

Comment:
You can move this conditional to method increaseSize() and avoid code duplicates